src/contained-list/contained-list-item.component.ts

```diff
@@ -0,0 +1,96 @@
+import {
+	ChangeDetectionStrategy,
+	Component,
+	EventEmitter,
+	HostBinding,
+	Input,
+	Output,
+	TemplateRef
+} from "@angular/core";
+
+@Component({
+	selector: "ibm-contained-list-item",
+	template: `
+		<ng-container *ngIf="clickable">
+			<button
+				class="cds--contained-list-item__content"
+				type="button"
+				[disabled]="disabled"
+				(click)="onClick()">
+				<ng-content select="[ibmContainedListItemButton]"></ng-content>
+			</button>
+		</ng-container>
+		<ng-container *ngIf="!clickable">
+			<div class="cds--contained-list-item__content">
+				<div *ngIf="icon" class="cds--contained-list-item__icon">
+					<ng-template [ngTemplateOutlet]="icon"></ng-template>
+				</div>
+				<ng-content></ng-content>
+			</div>
+		</ng-container>
+		<div class="cds--contained-list-item__action" *ngIf="action">
+			<ng-template [ngTemplateOutlet]="action"></ng-template>
+		</div>
+	`,
+	styles: [`
+		:host {
+			display: list-item;
+			list-style: none;
+		}
+	`],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ContainedListItem {
+	/**
+	 * A slot for a possible interactive element to render within the item.
+	 */
+	@Input() action: TemplateRef<any>;
+
+	/**
+	 * Whether this item is disabled.
+	 */
+	@Input() disabled = false;
+
+	/**
+	 * Whether this item is clickable.
+	 */
+	@Input() clickable: boolean;
+
+	/**
+	 * Provide an optional icon to render in front of the item's content.
+	 */
+	@Input() icon: TemplateRef<any>;
```

Comment:
```suggestion
	/**
	 * Provide an optional icon to render in front of the item's content.
	 */
	@Input() icon: TemplateRef<any> | string;
```

I thought about your original implementation with string @maicongodinho and I think we can make use of both! It's a really good idea in my opinion. What are your thoughts @zvonimirfras?

It would make it easier for teams that want to use ONLY carbon icons in their contained list. This would also save them the trouble of writing a `ng-template` 🤔

Reply:
If we do decide to go with this, we should add a comment letting users know that they need to register their `CARBON` icons.

Reply:
Yeah I agree, what is your thoughts on that @zvonimirfras?

Reply:
Let's do it!

Reply:
Check that @Akshat55 [43b7f43](https://github.com/carbon-design-system/carbon-components-angular/pull/2464/commits/43b7f432965b9cd5112d60e5ad00ce2c2008173f)